### PR TITLE
OSOE-124: Fixing VSTHRD111 and .NET SDK code analysis rules

### PR DESCRIPTION
**Pull request by Piedone**
OSOE-124

### CODE REVIEW

File: general.ruleset [hunk 1]
```diff
@@ -429,4 +430,4 @@
     <Rule Id="SX1101" Action="Warning" />
     <Rule Id="SX1309S" Action="Warning" />
   </Rules>
-</RuleSet>
+</RuleSet>
```
Comment (0liver) [inline general.ruleset:433]: Please don't remove empty lines at the end of files.

Comment (Piedone) [reply general.ruleset:433]: The VS graphical editor for rulesets removes it. Just as we changed the file name casing due to it constantly doing that, I wouldn't keep re-adding this just for the sake of it.

Comment (0liver) [reply general.ruleset:433]: I see. Makes sense, although it also makes me sad.

Comment (Piedone) [reply general.ruleset:433]: Me too.

**Review by sarahelsaig — APPROVED** [review]

**Pull request merged**